src/pages/Home.tsx

```
@@ -34,16 +34,37 @@ import { useMediaQuery } from "@/hooks/useMediaQuery";
+
   const [isMobileFilterOpen, setIsMobileFilterOpen] = useState(false);
   const [isDesktopSidebarOpen, setIsDesktopSidebarOpen] = useState(false);
 
   // Track if user has manually toggled viewMode
   const [manualViewMode, setManualViewMode] = useState<
     "compact" | "detailed" | null
   >(null);
+  const prefsAppliedRef = useRef<boolean | null>(null);
+
+  // Apply saved preferences once when they arrive (during render, not in effect)
+  if (preferences !== undefined && prefsAppliedRef.current == null) {
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/pages/Home.tsx:59:36
  57 |
  58 |   // Apply saved preferences once when they arrive (during render, not in effect)
> 59 |   if (preferences !== undefined && prefsAppliedRef.current == null) {
     |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Cannot access ref value during render
  60 |     prefsAppliedRef.current = true;
  61 |     if (preferences?.viewMode != null) {
  62 |       setManualViewMode(preferences.viewMode);

(react-hooks/refs)